assignments/array-methods.js

```diff
@@ -56,28 +56,75 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+runners.forEach((runner) => {
+    const name = {
+        firstName: runner.first_name,
+        lastName: runner.last_name
+    };
+    fullName.push(name);
+});
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+runners.map((runner) => {
```

Comment:
map will create a new array in place, you can just assign it to allCaps instead of pushing. Also, map doesn't require you to do anything to the whole object, just to portion you want to change.
```suggestion
let allCaps = runners.map(runner => {
  return runner.first_name.toUpperCase()
})
```

Reply:
ok.. thought we had to return the whole object with upper case name. I will change it.

Reply:
Oh, I see what you mean. It's fine, but if you do want to return the whole object, you can do this:
```suggestion
let allCaps = runners.map(runner => {
  runner.first_name = runner.first_name.toUpperCase()
  return runner
})
```
You're still only modifying the one variable. You don't have to write out the whole object.

Reply:
does that modify the original object through?

Reply:
Oh, wow. You're right. I should've double-checked. There is a shorter way than writing out the whole object, though. You can use Object.assign, like this:

```suggestion
runners.map(runner => {
      return Object.assign({}, runner, {name: runner.first_name.toUpperCase()})
    }
  )
```

or, you can use the spread operator (my personal favorite):

```suggestion
runners.map(runner => {
      return {
      ...runner,
      name: runner.first_name.toUpperCase()
      }
    }
  )
```
Sorry for the incorrect info.

Reply:
ooooooo, nice. Looks like object.assign can be useful because I was wondering how to overwrite just a single prop in an object while bringing the others as well...

that spread operator is very slick... that works for arrays too I think and assigning array values to variables. Thanks.